Fix crash in zone component load summary

JasonGlazer opened the pull request:
Fixes #5357

JasonGlazer committed:
Fixed bug which looks like it was a copy and paste without changing cool to heat.

JasonGlazer committed:
Added unit test within OutputReportTabular related to GetColumnUsingTabs which previously had not been getting tested.

JasonGlazer committed:
Merge branch 'develop' into 110568302-FixCrashInZoneComponentLoadSummary

JasonGlazer committed:
Merged latest develop.

JasonGlazer commented:
The only diffs seen are with 5ZoneFanCoilDOASCool.idf which are expected. Decided to merge to the latest version of develop since it seemed stale.

JasonGlazer commented:
After merge with develop the only diffs seen are still with 5ZoneFanCoilDOASCool.idf which are expected. This is ready for review and merge.

Myoldmopar commented on tst/EnergyPlus/unit/OutputReportTabular.unit.cc:
```diff
@@ -235,6 +235,29 @@ TEST( OutputReportTabularTest, GetUnitConversion )
+	EXPECT_EQ( "Col1", GetColumnUsingTabs( inString3, 1 ) );
+	EXPECT_EQ( "Col2", GetColumnUsingTabs( inString3, 2 ) );
+	EXPECT_EQ( "Col3", GetColumnUsingTabs( inString3, 3 ) );
+	EXPECT_EQ( "", GetColumnUsingTabs( inString3, 4 )  );
```
@JasonGlazer As I scan through this unit test, I'm thinking it might help robustness if we add a couple more possibilities, unless you think they aren't needed.  My thoughts are:
- `std::string inString4 = "";`
- `std::string inString5 = " ";`
- `std::string inString6 = "\t";`
- `std::string inString7 = "\tCol1\tCol2\tCol3\t";`
- `std::string inString8 = "Col1\t\tCol2\tCol3\t";`

Would you mind adding those cases in and making sure the function passes all those cases?  If so this should be good to go right in, and I'd really feel good we were exercises a broad spectrum of possible cases.

JasonGlazer replied:
Sure, although I am not sure the original code is going to handle these other cases so it might require additional fixes.

Myoldmopar replied:
Well, I think just being clear about what the function is expected to do and not is good.  You don't have to add support for all those, you could even put in purposely failing tests if that is easier.  If it's clear what this is supposed to cover, just make sure the function description captures that and we'll move on.

mbadams5 replied:
@Myoldmopar I added those new test cases and I think I captured the correct behavior of this function.

mbadams5 committed:
Merge branch 'develop' into 110568302-FixCrashInZoneComponentLoadSummary

mbadams5 committed:
Update unit tests and refactor GetColumnUsingTabs

I refactored GetColumnUsingTabs so that it is more efficient and more
c++ style.

I also added the extra unit tests Edwin suggested in his PR comment.

JasonGlazer commented:
Thanks @mbadams5 !

mbadams5 committed:
Missed a line in my commit somehow

Myoldmopar commented:
Ohhh, CI already started on 6e204b8.....so close.
